### PR TITLE
Remove deprecated `xpack.searchable.snapshot.allocate_on_rolling_restart` setting

### PR DESCRIPTION
**Pull request by arteam**
The setting was created as an escape-hatch in case elastic/elasticsearch#66369 had some unintended side-effects. It has been deprecated since elastic/elasticsearch#84959 (8.2.0).

We fully rely on the `cluster.routing.allocation.enable` setting for allocating searchable snapshots as specified in https://www.elastic.co/guide/en/elasticsearch/reference/current/restart-cluster.html#restart-cluster-rolling

### CODE REVIEW

**Comment by arteam** [comment]
@elasticmachine update branch

**Comment by arteam** [comment]
Hi @DaveCTurner! Would you able to get a look at the PR?

File: docs/changelog/114202.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+type: breaking
+issues: []
+breaking:
+  title: Remove deprecated `xpack.searchable.snapshot.allocate_on_rolling_restart` setting
+  area: 'Cluster and node setting'
+  details: >-
+    The `xpack.searchable.snapshot.allocate_on_rolling_restart` setting was created as an escape-hatch just in case
+    relying on the `cluster.routing.allocation.enable=primaries` setting for allocating searchable snapshots during
+    rolling restarts had some unintended side-effects. It has been deprecated since 8.2.0.
+  impact: No known end user impact
```
Comment (DaveCTurner) [inline docs/changelog/114202.yaml:13]:
```suggestion
  impact: Remove `xpack.searchable.snapshot.allocate_on_rolling_restart` from your settings if present.
```

**Review by DaveCTurner — APPROVED** [review]
LGTM

**Comment by arteam** [comment]
@elasticmachine update branch

**Comment by leemthompo** [comment]
@arteam is this PR relevant to the serverless changelog? [FYI this question is based on [9.0 breaking changes](https://staging-website.elastic.co/docs/release-notes/elasticsearch/breaking-changes)]

**Pull request merged**